packages/terra-application-layout/docs/usage.md

```diff
@@ -0,0 +1,264 @@
+
+## Menu/Content Component Configuration
+
+The `routingConfig` prop specifies which components will be rendered in the ApplicationLayout's `content` and `menu` regions. These components receive a set of props from the ApplicationLayout that they must utilize; otherwise, these components can be completely defined by the consuming application.
+
+> Note: Because the `ApplicationLayout` utilizes client-side routing provided by `react-router-dom`, the usage of `<a href=>` links and page anchors is restricted. Usage of links and page anchors may negatively impact navigation within an application. `<Link />` components provided by `react-router-dom` or custom events can and should be used instead.
+
+The components rendered in these regions will receive the following props from the ApplicationLayout:
+
+### `app`
+
+The AppDelegate instance (as `app`) provided to the ApplicationLayout's contents is created by ApplicationLayout's default `ModalManager`.
+
+Please review the AppDelegate documenation with `terra-app-delegate` for more information regarding the available disclosure options.
+
+### `layoutConfig`
+
+The `layoutConfig` contains properties describing the ApplicationLayout's state and functions that allow for the manipulation of that state.
+
+Please review the `Layout` component in `terra-layout` for more information.
+
+### `routingStackDelegate`
+
+The `routingStackDelegate` contains APIs that allow for virtual navigation within a set of matched component paths.
+
+Imagine a `routingConfiguration` that looks something like this:
+
+```jsx
+const myRoutingConfig = {
+  menu: {
+    '/a': {
+      path: '/a',
+      component: {
+        default: {
+          componentClass: AMenuComponent
+        }
+      }
+    },
+    '/a/b': {
+      path: '/a/b',
+      component: {
+        default: {
+          componentClass: BMenuComponent
+        }
+      }
+    },
+    '/a/b/c': {
+      path: '/a/b/c',
+      component: {
+        default: {
+          componentClass: CMenuComponent
+        }
+      }
+    },
+  }
+  content: {
+    '/a': {
+      path: '/a',
+      component: {
+        default: {
+          componentClass: AContentComponent
+        }
+      }
+    }
+  }
+}
+```
+
+If the global router location is `/a/b/c`, the ApplicationLayout will render `CMenuComponent` in the menu region and `AContentComponent` in the content region. However, given the nested nature of the configuration, the `BMenuComponent` and `AMenuComponent` may still provide important navigation or contextual information for the `AContentComponent`. It would be nice to be able to be able to view those menus without changing the global router location and losing the current users current application context.
```

Comment:
At this point we haven't done this in any docs, but I feel if we added basic visuals to these would help clarify these descriptions. I am thinking Before and After with menu & content labels with the global router highlighted in the browser.

Reply:
An idea could be to break this doc out into pieces and add a `slide-group` doc section that should visually navigate this example with the explanation.

Reply:
I agree, though at this point we may need to come back around on with an enhancement.

However, I will be making a reusable component (before merging) that people can use directly for their menu components. It will handle a lot of these complexities for them, which should help out a large number of consumers. It will also act as a good, maintained example of how to interact with these APIs for folks who roll their own solution.